docs/modules/ROOT/pages/tls.adoc

```diff
@@ -0,0 +1,57 @@
+= TLS
+
+A TLS section is part of Stackable CRDs and describes how to connect to a TLS enabled system like LDAP or S3.
+
+If the `tls` attribute is set to `null` (or is not specifed) no TLS will be used for the connection.
+
+A simple TLS section looks like this:
+
+[source,yaml]
+----
+include::example$tls-simple.yaml[]
+----
+
+== Verification
+The parties participating at an TLS connection can be verified using certificates.
+At the moment the following verification methods are supported
+
+* <<No verification>>
+* <<Server verification>>
+* <<Mutual verification>>
+
+=== No verification
+This example will use TLS but not check the authority of server or client:
+
+[source,yaml]
+----
+include::example$tls-no-verification.yaml[]
+----
+
+=== Server verification
+This example will use TLS and verify the server using the ca certificates that are trusted by common web browsers.
```

Comment:
This is not strictly speaking correct I think. This would trust anything that is signed by a CA which is present in the default ca store in the docker image running our products. I'd assume that this is empty by default. Customers should probably be able to specify a configmap to mount as ca store if they need their own CAs to be rolled out though..

Reply:
I'm not sure about adding CAs to our images by default tbh. Most customers I know are very restrictive about which CAs are allowed in production images, but I think that is not something that needs to stop this PR from going forward.